lib/potentialbuild.rb

```diff
@@ -936,10 +867,10 @@ def post_results(compiler, pending)
       end
 
       if !pending && @config.post_results_comment
-        if !@commit_sha.nil? && @repository == @config.repository
-          github_query(@client) { @client.create_commit_comment(@config.repository, @commit_sha, github_document) }
-        elsif !@pull_id.nil?
+        if !@pull_id.nil?
```

Comment:
Here I am now prioritizing the PR ID check when deciding where to write the comment about Decent results.  In this way, whether the build comes from an external or internal PR, it should have a valid pull_id, and so this will write it right on the PR page.  And then if it is coming from a branch without a PR, it will fall through to the ELSE and just write it on the commit itself.